ext/lib/crypto/mbedtls/include/mbedtls/platform.h

```diff
@@ -223,7 +223,7 @@ int mbedtls_platform_set_snprintf( int (*snprintf_func)( char * s, size_t n,
 #if defined(MBEDTLS_PLATFORM_SNPRINTF_MACRO)
 #define mbedtls_snprintf   MBEDTLS_PLATFORM_SNPRINTF_MACRO
 #else
-#define mbedtls_snprintf   MBEDTLS_PLATFORM_STD_SNPRINTF
+#define mbedtls_snprintf   snprintf
```

Comment:
this looks wrong. In the beginning of this file we have:

```
#if !defined(MBEDTLS_PLATFORM_NO_STD_FUNCTIONS)
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#if !defined(MBEDTLS_PLATFORM_STD_SNPRINTF)
#if defined(_WIN32)
#define MBEDTLS_PLATFORM_STD_SNPRINTF   mbedtls_platform_win32_snprintf /**< The default \c snprintf function to use.  */
#else
#define MBEDTLS_PLATFORM_STD_SNPRINTF   snprintf /**< The default \c snprintf function to use.  */
#endif
#endif
```

so either you are previously defining `MBEDTLS_PLATFORM_STD_SNPRINTF` or `MBEDTLS_PLATFORM_NO_STD_FUNCTIONS`, otherwise this macro would expand to `snprintf`.

Reply:
@ceolin : In this case, MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is being declared.

If this patch is not OK, would you mind considering this alternative https://github.com/JuulLabs-OSS/mcuboot/pull/436/commits/eb2dc14d22dddc11600e66dd489655e6362e7be0

?